app/src/main/java/com/example/pokemonapp/di/PokemonAppModule.kt

```diff
@@ -0,0 +1,37 @@
+package com.example.pokemonapp.di
+
+import com.example.pokemonapp.data.dataSource.PokemonRemoteDataSource
+import com.example.pokemonapp.data.dataSource.PokemonRemoteDataSourceImpl
+import com.example.pokemonapp.data.repository.PokemonRepository
+import com.example.pokemonapp.data.repository.PokemonRepositoryImpl
+import com.example.pokemonapp.network.PokemonApiService
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+
+@Module
+@InstallIn(SingletonComponent::class)
+object PokemonAppModule {
+
+    @Singleton
+    @Provides
+    fun providePokemonRepository(pokemonRemoteDataSource: PokemonRemoteDataSource): PokemonRepository =
+        PokemonRepositoryImpl(pokemonRemoteDataSource)
+
+}
+
+
+@Module(includes = [NetworkModule::class])
```

Comment:
Can you explain this

Reply:
to call a module inside another module, but it changed and not needed anymore since i changed the module form object to class